lib/lightning/projects.ex

```diff
@@ -169,6 +172,31 @@ defmodule Lightning.Projects do
     |> Repo.all()
   end
 
+  def insert_with_global_oauth_clients(changeset) do
+    case Repo.insert(changeset) do
+      {:ok, project} ->
+        associate_global_oauth_clients(project)
+        {:ok, project}
+
+      error ->
+        error
+    end
+  end
+
+  defp associate_global_oauth_clients(%Project{id: project_id}) do
+    # Fetch global OAuth clients
+    global_clients = Repo.all(from c in OauthClient, where: c.global)
+
+    # Associate each global OAuth client with the new project
+    Enum.each(global_clients, fn %OauthClient{id: oauth_client_id} ->
+      %ProjectOauthClient{
+        oauth_client_id: oauth_client_id,
+        project_id: project_id
+      }
+      |> Repo.insert()
+    end)
+  end
```

Comment:
@elias-ba @christad92 I'm not sure I understand this, if a superuser creates a global oauth client, every project that is made from that moment onward will get it associated with it?

Even clients/projects via self service. That sounds crazy, I must be missing something here.

Reply:
That aside, what about older projects, they won't inherit these oauth clients - unless there is some kind of migration function I'm not aware of.

Reply:
This is fixed now. We are no longer creating project_oauth_clients record for global clients. We are just handling those in select queries by doing unions. Thanks